quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java

```diff
@@ -0,0 +1,187 @@
+package com.rl.mpquoridor.models;
+
+import com.rl.mpquoridor.enums.Cell;
+import com.rl.mpquoridor.enums.Direction;
+import lombok.Data;
+
+import java.util.LinkedList;
+
+@Data
+public class Board {
+    private Cell[][] board;
+
+    public Board(int size, Position[] pawnPositions) {
+        int newSize = size * 2 - 1;
+        this.board = new Cell[newSize][newSize];
+        this.initBoard();
+
+        for (Position position: pawnPositions) {
+            this.board[position.getY() * 2][position.getX() * 2] = Cell.PAWN;
+        }
+    }
+
+    public void printBoard() {
+        for (int height = 0; height < this.board.length; height++) {
+            System.out.println("");
+            for (int width = 0; width < this.board[0].length; width++) {
+                System.out.print(this.board[height][width] + " , ");
+            }
+        }
+
+        System.out.println("");
+    }
+
+    public boolean movePawn(Position fromPosition, Position toPosition) {
+        //TODO: check if this is a valid new position
+        this.board[fromPosition.getY() * 2][fromPosition.getX() * 2] = Cell.EMPTY;
+        this.board[toPosition.getY() * 2][toPosition.getX() * 2] = Cell.PAWN;
+        return true;
+    }
+
+    public boolean addWall(Wall wall, Position[] positions, int[] heightsWin) {
+        if (this.checkAddingWall(wall, positions, heightsWin)) {
+            this.board = this.addWallToBoard(wall, this.board);
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean checkAddingWall(Wall wall, Position[] positions, int[] heightsWin) {
+        Direction direction = wall.getDirection();
+        Position position = wall.getPosition();
+        int newX = position.getX() * 2 + 1;
+        int newY = position.getY() * 2 + 1;
+
+        if (direction == Direction.Right) {
```

Comment:
with enums the best practice is to use ".equals" and not == (also in strings and other objects)